property_tests/arbitraries/candid/reference/service_arb/definition_arb.ts

```diff
@@ -6,60 +6,81 @@ import {
 } from './service_method_arb';
 import {
     CandidDefinition,
-    ServiceCandidDefinition
+    ServiceCandidDefinition,
+    WithShapes,
+    WithShapesArb
 } from '../../candid_definition_arb/types';
 import { UniqueIdentifierArb } from '../../../unique_identifier_arb';
 import { Canister } from '../../../../../src/lib/candid/types/reference/service';
 
 export function ServiceDefinitionArb(
-    fieldCandidDefArb: fc.Arbitrary<CandidDefinition>
-): fc.Arbitrary<ServiceCandidDefinition> {
+    fieldCandidDefArb: WithShapesArb<CandidDefinition>
+): WithShapesArb<ServiceCandidDefinition> {
     return fc
         .tuple(
             UniqueIdentifierArb('typeDeclaration'),
             fc.uniqueArray(ServiceMethodArb(fieldCandidDefArb), {
-                selector: (entry) => entry.name
+                selector: (entry) => entry.definition.name
             }),
             fc.boolean()
         )
-        .map(([name, fields, useTypeDeclaration]): ServiceCandidDefinition => {
-            useTypeDeclaration = true;
-
-            const candidTypeAnnotation = generateCandidTypeAnnotation(
-                useTypeDeclaration,
-                name
-            );
-
-            const candidTypeObject = generateCandidTypeObject(
-                useTypeDeclaration,
-                name,
-                fields
-            );
-
-            const runtimeCandidTypeObject =
-                generateRuntimeCandidTypeObject(fields);
-
-            const variableAliasDeclarations = generateVariableAliasDeclarations(
-                useTypeDeclaration,
-                name,
-                fields
-            );
-
-            const imports = generateImports(fields);
-
-            return {
+        .map(
+            ([
                 name,
-                candidMeta: {
-                    candidTypeAnnotation,
-                    candidTypeObject,
-                    runtimeCandidTypeObject,
-                    variableAliasDeclarations,
-                    imports,
-                    candidType: 'Service'
-                },
-                funcs: fields
-            };
-        });
+                fieldsAndShapes,
+                useTypeDeclaration
+            ]): WithShapes<ServiceCandidDefinition> => {
+                const fields = fieldsAndShapes.map(
+                    (fieldAndShapes) => fieldAndShapes.definition
+                );
+                const recursiveShapes = fieldsAndShapes.reduce(
+                    (acc, fieldAndShapes) => {
+                        return { ...acc, ...fieldAndShapes.recursiveShapes };
+                    },
+                    {}
+                );
+                useTypeDeclaration = true;
```

Comment:
Let's not have this mutation thing

Reply:
Apparently we needed this for service to work. It was not added as part of this pr, it just looks like that since the formatting changed